src/platform/cannonlake/include/platform/clk.h

```diff
@@ -33,10 +33,33 @@
 #ifndef __PLATFORM_CLOCK__
 #define __PLATFORM_CLOCK__
 
-#define CLK_CPU		0
-#define CLK_SSP		1
+#include <sof/cpu.h>
+#include <sof/io.h>
+
+#define CLK_CPU(x)	(x)
+#define CLK_SSP		4
+
+#define CPU_DEFAULT_IDX		1
+#define SSP_DEFAULT_IDX		1
 
 #define CLK_DEFAULT_CPU_HZ	120000000
 #define CLK_MAX_CPU_HZ		400000000
 
+#define NUM_CLOCKS	5
+
+static inline int clock_platform_set_cpu_freq(uint32_t cpu_freq_enc)
+{
+	/* set CPU frequency request for CCU */
+	io_reg_update_bits(SHIM_BASE + SHIM_CLKCTL,
+			   SHIM_CLKCTL_DPCS_MASK(cpu_get_id()),
+			   cpu_freq_enc);
+
+	return 0;
+}
+
+static inline int clock_platform_set_ssp_freq(uint32_t ssp_freq_enc)
+{
+	return 0;
```

Comment:
Should this not return an error is any frequency other than supported frequency is requested ?

Reply:
@lgirdwood It is handled before calling this function, since it searches for the most valid frequency from the set of platform supported frequencies.

Reply:
@tlauda ok, so best put a comment in these so it's not ambiguous.